text/0000-data-formats.md

```diff
@@ -0,0 +1,151 @@
+# RFC: Data Formats for OpenTelemetry
+
+# Overview
+OpenTelemetry is an observability system, which generates observations, and then sends that data to remote systems for analysis. Describing how  this observability data is represented and transmitted is critical part of the OpenTelemetry specification.
+
+**Proposed Layers**
+*   Metadata and Semantic Conventions
+*   Logical Structure
+*   Encodings
+*   Exchange Protocols
+
+
+# Metadata and Semantic Conventions
+
+Spans represent specific operations in and between systems. Some of these operations represent calls that use well-known protocols like HTTP or database calls. Depending on the protocol and the type of operation, additional information is needed to represent and analyze a span correctly in monitoring systems. It is also important to unify how this attribution is made in different languages. This way, the operator will not need to learn specifics of a language and telemetry collected from multi-language micro-service can still be easily correlated and cross-analyzed.
+
+## Requirements
+
+
+### Handle complex data
+
+Semantic information is often complex and multi-valued. How we represent semantic data is related to how attributes are defined at the logical layer.
+
+*   At minimum, nested lists of key/value pairs are necessary to represent them.
+*   Is it advantageous to also have strongly typed data structures?
+*   Are there issues with having strongly typed data structures in certain environment, such as dynamic languages?
+
+
+### Compose Multiple Semantics
+
+Objects in OpenTelemetry may describe more than one type of operation. For example, a span may simultaneously represent a MySQL transaction and a TCP/IP request. Pieces of metadata must compose well, without losing the structure integrity of each piece or risking a conflict in meaning.
+
+
+### Easily Extensible
+
+Not all semantics conventions and common operations are known at this time. It must be possible to add more semantic definitions without creating to versioning or compatibility issues with OpenTelemetry implementations.
+
+
+## Examples of Common Operations
+
+#### RPC calls
+
+Modeling distributed transactions is a core usecase, it is common to analyze  errors and latency for RPC calls. Because multiple individual network requests may be involved in a single logical RPC operation, it must be possible to describe RPC calls as logical operations, independent from the descriptions of transport and application protocols which they contain.
```

Comment:
```suggestion
Modeling distributed transactions is a core usecase, it is common to analyze errors and latency for RPC calls. Because multiple individual network requests may be involved in a single logical RPC operation, it must be possible to describe RPC calls as logical operations, independent from the descriptions of transport and application protocols which they contain.
```